# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1313602119 -7200
#      Wed Aug 17 19:28:39 2011 +0200
# Node ID 83bcf326e9849184c8e3768510e217fd6afee9bf
# Parent  6ab76213ea748600f33d84e8ce3f0328a955b24c
Completion of usernames in channel works.

diff --git a/gui/src/IRCWidget.cpp b/gui/src/IRCWidget.cpp
--- a/gui/src/IRCWidget.cpp
+++ b/gui/src/IRCWidget.cpp
@@ -47,17 +47,17 @@ ChatMessageTextEdit::setCompleter (QComp
     QObject::disconnect (m_completer, 0, this, 0);
 
   m_completer = completer;
 
   if (!m_completer)
     return;
 
   m_completer->setWidget (this);
-  m_completer->setCompletionMode (QCompleter::InlineCompletion);
+  m_completer->setCompletionMode (QCompleter::PopupCompletion);
   m_completer->setCaseSensitivity (Qt::CaseInsensitive);
   QObject::connect (m_completer, SIGNAL (activated (QString)),
                     this, SLOT (insertCompletion (QString)));
 }
 
 QCompleter *
 ChatMessageTextEdit::completer () const
 {
@@ -118,19 +118,30 @@ ChatMessageTextEdit::keyPressEvent (QKey
       return;
     default:
       QPlainTextEdit::keyPressEvent(keyPressEvent);
       break;
       }
 
     QString completionPrefix = textUnderCursor ();
     if (completionPrefix != m_completer->completionPrefix ())
-      m_completer->setCompletionPrefix(completionPrefix);
-    m_completer->complete ();
-   }
+      {
+        m_completer->setCompletionPrefix(completionPrefix);
+      }
+
+    if (completionPrefix.length() > 0)
+      {
+        m_completer->popup ()->setCurrentIndex (m_completer->completionModel ()->index (0, 0));
+        m_completer->complete ();
+      }
+    else
+      {
+        m_completer->popup ()->hide ();
+      }
+  }
 }
 
 IRCWidget::IRCWidget (QWidget * parent):
 QWidget (parent)
 {
   QSettings *settings = ResourceManager::instance ()->settings ();
   bool connectOnStartup = settings->value ("connectOnStartup").toBool ();
   m_autoIdentification = settings->value ("autoIdentification").toBool ();
